### PR TITLE
Map Tile Generation

### PR DESCRIPTION
**Pull request by raigex**
Pull base url from MapFloor object

### CODE REVIEW

File: db/src/main/kotlin/edu/artic/db/models/ArticAppData.kt [hunk 1]
```diff
@@ -14,7 +14,7 @@ data class ArticAppData(
         @Json(name = "audio_files") val audioFiles: Map<String, ArticAudioFile>?,
         @Json(name = "tours") val tours: List<ArticTour>?,
         @Json(name = "map_annontations") val mapAnnotations: Map<String, ArticMapAnnotation>?,
-        @Json(name = "map_floors ") val mapFloors: Map<String, ArticMapFloor>,
+        @Json(name = "map_floors") val mapFloors: Map<String, ArticMapFloor>,
```
Comment (Cliabhach) [inline db/src/main/kotlin/edu/artic/db/models/ArticAppData.kt]: :100:

File: map/src/main/kotlin/edu/artic/map/MapViewModel.kt [hunk 2]
```diff
@@ -53,8 +57,9 @@ class MapViewModel @Inject constructor(val mapMarkerConstructor: MapMarkerConstr
     val leftActiveTour: Subject<Boolean> = PublishSubject.create()
     val switchTourRequest: Subject<Pair<ArticTour, ArticTour.TourStop>> = PublishSubject.create()
 
-    val distinctFloor: Observable<Int>
-        get() = floor.distinctUntilChanged()
+    private val distinceFloorInt = floor.distinctUntilChanged().filter { it in 0..3 }
```
Comment (Cliabhach) [inline map/src/main/kotlin/edu/artic/map/MapViewModel.kt]: `distince` -> `distinct`

File: map/src/main/kotlin/edu/artic/map/MapViewModel.kt [hunk 3]
```diff
@@ -65,7 +70,25 @@ class MapViewModel @Inject constructor(val mapMarkerConstructor: MapMarkerConstr
     private val lockVisibleRegion: Subject<Boolean> = BehaviorSubject.createDefault(false)
 
     init {
-        mapMarkerConstructor.bindToMapChanges(distinctFloor,
+        articMapFloorMap.subscribe {
+            Timber.d("mapsize: ${it.size}")
```
[the remark this answers is not in the document]
Comment (raigex) [reply map/src/main/kotlin/edu/artic/map/MapViewModel.kt]: sorry this was not supposed to stay there. was only there cuz I was debugging something.

File: map/src/main/kotlin/edu/artic/map/MapViewModel.kt [hunk 4]
```diff
@@ -65,7 +70,25 @@ class MapViewModel @Inject constructor(val mapMarkerConstructor: MapMarkerConstr
...
+
+        mapFloorDao.getMapFloors()
+                .map { floorMap -> floorMap.associateBy { it.label } }
+                .bindTo(articMapFloorMap)
+                .disposedBy(disposeBag)
+
+        Observables
+                .combineLatest(distinceFloorInt, articMapFloorMap)
+                .subscribeBy { (floor, floorMap) ->
+                    floorMap[floor.toString()]?.let {
```
[the remark this answers is not in the document]
Comment (raigex) [reply map/src/main/kotlin/edu/artic/map/MapViewModel.kt]: So i am weary of using Int's as keys for maps. but I'll update @agrosner

[the remark this answers is not in the document]
Comment (agrosner) [reply map/src/main/kotlin/edu/artic/map/MapViewModel.kt]: why? what's wrong with `Int`?

File: map/src/main/kotlin/edu/artic/map/MapViewModel.kt (hunk 2 above)
Comment (raigex) [reply map/src/main/kotlin/edu/artic/map/MapViewModel.kt]: done

File: map/src/main/kotlin/edu/artic/map/MapFragment.kt [hunk 5]
```diff
@@ -321,12 +275,11 @@ class MapFragment : BaseViewModelFragment<MapViewModel>() {
                 .disposedBy(disposeBag)
 
         viewModel.distinctFloor
-                .withLatestFrom(groundOverlayGenerated)
                 .observeOn(AndroidSchedulers.mainThread())
-                .filter { (floor, generated) -> generated && floor in 0..3 }
-                .withLatestFrom(viewModel.currentMap.filterValue()) { (floor), map -> floor to map }
+                .filter { floor -> floor.label.toInt() in 0..3 }
```
[the remark this answers is not in the document]
Comment (raigex) [reply map/src/main/kotlin/edu/artic/map/MapFragment.kt]: done

**Comment by Cliabhach** [comment]
It seems to run ok. I'll let @agrosner have the final word though

File: map/src/main/kotlin/edu/artic/map/MapViewModel.kt [hunk 6]
```diff
@@ -65,7 +70,25 @@ class MapViewModel @Inject constructor(val mapMarkerConstructor: MapMarkerConstr
     private val lockVisibleRegion: Subject<Boolean> = BehaviorSubject.createDefault(false)
 
     init {
-        mapMarkerConstructor.bindToMapChanges(distinctFloor,
+        articMapFloorMap.subscribe {
+            Timber.d("mapsize: ${it.size}")
+        }.disposedBy(disposeBag)
+
+        mapFloorDao.getMapFloors()
+                .map { floorMap -> floorMap.associateBy { it.label } }
```
[the remark this answers is not in the document]
Comment (raigex) [reply map/src/main/kotlin/edu/artic/map/MapViewModel.kt]: done

File: map/src/main/kotlin/edu/artic/map/MapViewModel.kt (hunk 4 above)
[the remark this answers is not in the document]
Comment (raigex) [reply map/src/main/kotlin/edu/artic/map/MapViewModel.kt]: done

File: map/src/main/kotlin/edu/artic/map/MapViewModel.kt [hunk 7]
```diff
@@ -65,7 +70,25 @@ class MapViewModel @Inject constructor(val mapMarkerConstructor: MapMarkerConstr
...
+        Observables
+                .combineLatest(distinceFloorInt, articMapFloorMap)
+                .subscribeBy { (floor, floorMap) ->
+                    floorMap[floor.toString()]?.let {
+                        distinctFloor.onNext(it)
+                    }
+
+                }.disposedBy(disposeBag)
+
+        mapMarkerConstructor.bindToMapChanges(distinctFloor.map { it.label.toInt() },
```
[the remark this answers is not in the document]
Comment (raigex) [reply map/src/main/kotlin/edu/artic/map/MapViewModel.kt]: done

**Pull request merged**